admin/securedrop_admin/__init__.py

```diff
@@ -667,29 +668,33 @@ def update(args):
 
     sdlog.info("Verifying signature on latest update...")
 
-    try:
-        # First try to get the release key using Tails default keyserver
-        get_release_key_from_keyserver(args)
-    except subprocess.CalledProcessError:
-        # Now try to get the key from a secondary keyserver.
-        secondary_keyserver = 'hkps://hkps.pool.sks-keyservers.net'
-        get_release_key_from_keyserver(args,
-                                       keyserver=secondary_keyserver)
+    # Retrieve key from openpgp.org keyserver
+    get_release_key_from_keyserver(args,
+                                   keyserver=DEFAULT_KEYSERVER)
 
     git_verify_tag_cmd = ['git', 'tag', '-v', latest_tag]
     try:
         sig_result = subprocess.check_output(git_verify_tag_cmd,
                                              stderr=subprocess.STDOUT,
                                              cwd=args.root)
 
-        good_sig_text = 'Good signature from "SecureDrop Release Signing Key"'
+        good_sig_text = ['Good signature from "SecureDrop Release Signing ' +
+                         'Key"',
+                         'Good signature from "SecureDrop Release Signing ' +
+                         'Key <securedrop-release-key@freedom.press>"']
         bad_sig_text = 'BAD signature'
-        # To ensure that an adversary cannot name a malicious key good_sig_text
-        # we check that bad_sig_text does not appear and that the release key
-        # appears on the second line of the output.
         gpg_lines = sig_result.split('\n')
+
+        # Check if any strings in good_sig_text match against gpg_lines[]
+        good_sig_matches = [s for s in gpg_lines if
```

Comment:
This is looking good - what do you think about doing `good_sig_matches = set(gpg_lines).intersection(good_sig_text)` here for readability (and leaning on the standard library to do the intersection)?  `good_sig_text` can remain a list as `set.intersection` can take any iterable

Reply:
I did this originally, but it means good_sig_text[] has to contain the entire line for each element, which kindof changes how it behaves. Maybe for the better, but my concern was that it might be brittle if the gpg output varies a little.

Reply:
hmm yeah this is a good point, ok let's leave as is